spec/rspec/rails/example/rails_example_group_spec.rb

```diff
@@ -56,5 +56,85 @@ class CurrentSample < ActiveSupport::CurrentAttributes
         group.run(failure_reporter) ? true : failure_reporter.exceptions
       ).to be true
     end
+
+    context 'with suite-level around-example hooks configured', if: ::Rails::VERSION::MAJOR >= 7 do
+      let(:uniquely_identifiable_metadata) do
+        { configured_around_example_hook: true }
+      end
+
+      class CurrentAttrsBetweenHooks < ActiveSupport::CurrentAttributes
+        attribute :request_id
+      end
+
+      # We have to modify the suite's around-each in RSpec.config, but don't
+      # want to pollute other tests with this (whether or not it is harmless
+      # to do so). There being no public API to read or remove hooks, instead
+      # it's necessary use some private APIs to be able to delete the added
+      # hook via 'ensure'.
+      #
+      around :each do | example |
+
+        # Client code might legitimately want to wrap examples to ensure
+        # all-conditions tidy-up, e.g. "ActsAsTenant.without_tenant do...",
+        # wherein an "around" hook is the only available solution, often used
+        # in the overall suite via "config.around". Tests would not expect
+        # anything set in CurrentAttributes here to suddenly be reset by the
+        # time their actual tests, or their test hooks ran.
+        #
+        RSpec.configure do | config |
+          config.around(:each, uniquely_identifiable_metadata()) do | example |
```

Comment:
You can use [in_sub_process](https://github.com/rspec/rspec-support/blob/23b3756f27ba0a506aa014dfd3e310c2bcd5f0a3/lib/rspec/support/spec/in_sub_process.rb#L12) to avoid config leaks between examples.

To differentiate between our examples and examples under test, we usually `let(:example_group) { RSpec.describe("test example group") }` in specs.

Otherwise, 👍

Reply:
Had no idea about `in_sub_process`. About to head home for the day but I'll look into that soon as it's likely to be very much cleaner. Thanks!

Reply:
...Actually, that was easy. Change pushed already.